apps/backend/src/modules/points-of-interest/points-of-interest.repository.ts

```diff
@@ -0,0 +1,88 @@
+import { type Repository } from "~/libs/types/types.js";
+import { PointsOfInterestEntity } from "~/modules/points-of-interest/points-of-interest.entity.js";
+import { type PointsOfInterestModel } from "~/modules/points-of-interest/points-of-interest.model.js";
+
+class PointsOfInterestRepository implements Repository {
+	private pointsOfInterestModel: typeof PointsOfInterestModel;
+
+	public constructor(pointsOfInterestModel: typeof PointsOfInterestModel) {
+		this.pointsOfInterestModel = pointsOfInterestModel;
+	}
+
+	public async create(
+		entity: PointsOfInterestEntity,
+	): Promise<PointsOfInterestEntity> {
+		const { name } = entity.toNewObject();
+
+		const pointOfInterest = await this.pointsOfInterestModel
+			.query()
+			.insert({
+				name,
+			})
+			.returning("*")
+			.execute();
+
+		return PointsOfInterestEntity.initialize(pointOfInterest);
+	}
+
+	public async delete(id: number): Promise<boolean> {
+		const deletedCount = await this.pointsOfInterestModel
+			.query()
+			.deleteById(id)
+			.execute();
+
+		return !!deletedCount;
```

Comment:
let's prefer a more explicit cast instead of !!, just a common pattern used in the app
```suggestion
		return Boolean(deletedCount);
```